packages/core/src/browser/shell/application-shell.ts

```diff
@@ -587,6 +588,17 @@ export class ApplicationShell extends Widget {
         return pinned;
     }
 
+    // Get an array corresponding to bottom panel widgets' pinned state.
+    getPinnedBottomWidgets(): boolean[] {
+        const pinned: boolean[] = [];
+
+        toArray(this.bottomPanel.widgets()).forEach((a, i) => {
+            pinned[i] = a.title.className.indexOf('theia-mod-pinned') >= 0;
+        });
+
+        return pinned;
+    }
```

Comment:
Not sure that this warrants its own function. Maybe `getPinnedWidgetsForArea(area)` that can handle both of them?